internal/test/types/types.go

```diff
@@ -0,0 +1,271 @@
+//		jira: https://issues.redhat.com/browse/WINC-239. Remove the fields related to SSHClient and WinrmClient and put
+// 		them in connectivity struct
+type Windows struct {
+	// Creds is used for parsing the vmCreds command line argument
+	Credentials *Credentials
+	// SSHClient contains the ssh client information to access the Windows VM via ssh
+	SSHClient *ssh.Client
+	// WinrmClient to access the Windows VM created
+	WinrmClient *winrm.Client
+}
+
+// WindowsVM is the interface for interacting with a Windows object created by the cloud provider
+type WindowsVM interface {
+	// CopyFile copies the given file to the remote directory in the Windows VM. The remote directory is created if it
+	// does not exist
+	CopyFile(string, string) error
+	// Run executes the given command remotely on the Windows VM and returns the output of stdout and stderr. If the
+	// bool is set, it implies that the cmd is to be execute in PowerShell.
+	Run(string, bool) (string, string, error)
+	// Run executes the given command remotely on the Windows VM over a ssh connection and returns the combined output
+	// of stdout and stderr. If the bool is set, it implies that the cmd is to be execute in PowerShell. This function
+	// should be used in scenarios where you want to execute a command that runs in the background. In these cases we
+	// have observed that Run() returns before the command completes and as a result killing the process.
+	RunOverSSH(string, bool) (string, error)
+	// GetCredentials returns the interface for accessing the VM credentials. It is up to the caller to check if non-nil
+	// Credentials are returned before usage.
+	GetCredentials() *Credentials
+	// Reinitialize re-initializes the Windows VM. Presently only the ssh client is reinitialized.
+	Reinitialize() error
+}
+
+func (w *Windows) CopyFile(filePath, remoteDir string) error {
+	if w.SSHClient == nil {
+		return fmt.Errorf("CopyFile cannot be called without a SSH client")
+	}
+
+	ftp, err := sftp.NewClient(w.SSHClient)
+	if err != nil {
+		return fmt.Errorf("sftp client initialization failed: %v", err)
+	}
+	defer ftp.Close()
+
+	f, err := os.Open(filePath)
+	if err != nil {
+		return fmt.Errorf("error opening %s file to be transferred: %v", filePath, err)
+	}
+	defer f.Close()
+
+	if err = ftp.MkdirAll(remoteDir); err != nil {
+		return fmt.Errorf("error creating remote directory %s: %v", remoteDir, err)
+	}
+
+	remoteFile := remoteDir + "\\" + filepath.Base(filePath)
+	dstFile, err := ftp.Create(remoteFile)
+	if err != nil {
+		return fmt.Errorf("error initializing %s file on Windows VMs: %v", remoteFile, err)
+	}
+
+	_, err = io.Copy(dstFile, f)
+	if err != nil {
+		return fmt.Errorf("error copying %s to the Windows VM: %v", filePath, err)
+	}
+
+	// Forcefully close it so that we can execute the binary later
+	dstFile.Close()
+	return nil
+}
+
+func (w *Windows) Run(cmd string, psCmd bool) (string, string, error) {
```

Comment:
Now I see why WinRM is required. Please have a single `Run` method that uses SSH like we did in WMCO.